Localization/extended_kalman_filter/extended_kalman_filter.py

```diff
@@ -147,9 +147,9 @@ def plot_covariance_ellipse(xEst, PEst):  # pragma: no cover
     b = math.sqrt(eigval[smallind])
     x = [a * math.cos(it) for it in t]
     y = [b * math.sin(it) for it in t]
-    angle = math.atan2(eigvec[bigind, 1], eigvec[bigind, 0])
-    rot = np.array([[math.cos(angle), math.sin(angle)],
-                    [-math.sin(angle), math.cos(angle)]])
+    angle = math.atan2(eigvec[1, bigind], eigvec[0, bigind])
```

Comment:
I noticed that particle filter is using different indexing.
https://github.com/AtsushiSakai/PythonRobotics/blob/468be047aa5975265cbdc5b89740bc6b61079a9e/Localization/particle_filter/particle_filter.py#L191

Do you think which is better?

Reply:
I think the angle calculated by the original index is inverse to the actual angle because the original index points to $(v2_x, v1_x)$, instead of $(v1_y, v1_x)$.

```
angle = math.atan2(eig_vec[big_ind, 1], eig_vec[big_ind, 0])
# ---> atan2(v2_x, v1_x) = atan2(-v1_y, v1_x) = -atan2(v1_y, v1_x) = -theta
```

It was ok in `extended_kalman_filter.py` and `unscented_kalman_filter.py` because the rotation matrix was also inverse. 

```
rot = np.array([
    [math.cos(angle), math.sin(angle)],
    [-math.sin(angle), math.cos(angle)]
])
# ---> R^T(-theta) = R^(-1)(-theta) = R(theta)
```

However, in `particle_filter.py` (and `ensemble_kalman_filter.py` as well), this inverse angle would cause a problem because the rotation matrix is now calculated by `scipy.spatial.transfor.Rotation.from_euler('z', angle)`.

I've tested the original `plot_covariance_ellipse()` function with a simple covariance matrix.
The angle should be +45.0 [deg] because the $sigma_xy$ is positive (in this case $sigma_xy$ == 2), but the output angle was -45.0 [deg].

![Screen Shot 2020-09-27 at 20 31 43](https://user-images.githubusercontent.com/26198619/94364631-ca929380-0105-11eb-9621-c11eb331b365.png)

Reply:
Thank you for checking. These are bugs I think. Could you please fix `particle_filter.py` and `ensemble_kalman_filter.py` ? After the fix these, I will merge this PR.

Reply:
Ok, I fixed it in [4cc0dea](https://github.com/AtsushiSakai/PythonRobotics/pull/407/commits/4cc0dea8bd76b48730c0684760623b74cac28a7c)

Reply:
Thank you for reviewing the code.

Reply:
Thank you too.